docs/getting-started/auth.adoc

```diff
@@ -1,15 +1,169 @@
+
+Callback authCallback = new Callback<UserPrincipal>() {
+    @Override
+    public void onSuccess(UserPrincipal principal) {
+        // User authenticated in, continue on..
+    }
+
+    @Override
+    public void onError(Throwable error) {
+        // An error occurred during login.
+    }
+}
+
+authService.login(options, authCallback);
+----
+
+Once the browser returns to the app the result can be handled. In the activity provided to the
+`login` method override `onActivityResult`. This handler should always invoke
+`AuthService#handleAuthResponse`, providing the `Intent`. This will exchange the temporary tokens
+returned from `AuthService#login` for long-life tokens and will provide a
+`UserPrincipal` which can be used to access a users details.
+
+More information about the user returned is available in link:../core/README.adoc[the auth module JavaDocs].
+
+[source,java]
+----
+@Override
+public void onActivityResult(int requestCode, int resultCode, Intent data) {
+    if (requestCode == LOGIN_RESULT_CODE) {
+        // The core will return the same instance of the auth service as before
+        AuthService authService = mobileCore.getInstance(AuthService.class);
+        authService.handleAuthResponse(data);
+    }
+}
+----
+
+The callback provided in `AuthService#login` will be invoked.
+
+=== Retrieving a users roles
+
+Once a `UserPrincipal` has been retrieved the roles of the user can be listed and checked. This can
+be used to perform client side access control, such as hiding UI components related to actions the
+user doesn't have permissions to perform.
+
+To list a users roles the `UserPrincipal#getRoles` method can be invoked.
+
+Roles are divided into two types. Client roles which belong to the client the user has
+authenticated against, and Realm roles which belong to the realm the client is in.
+
+In order to check if a user has a specific role you can invoke the `UserPrincipal#hasClientRole`
+and `UserPrincipal#hasRealmRole` methods and provide the role name to check for.
+
+[source,java]
+----
+// authService already initialized.
+AuthService authService = mobileCore.getInstance(AuthService.class);
+UserPrincipal currentUser = authService.currentUser();
+
+boolean hasAdminPermissions = currentUser.hasRealmRole("user_admin");
+if (hasAdminPermissions) {
+    // Show some component.
+}
+
+// Check if a user has a role from a specific client named my_client.
+boolean isModerator = currentUser.hasClientRole("my_client", "user_moderator");
+if (isModerator) {
+    // Enable some button.
+}
+----
+
+=== Logging out
+
+To logout, invoke the `AuthService#logout` method. This accepts the `UserPrincipal` that was
+provided by `AuthService#handleAuthResponse`.
+
+[source,java]
+----
+// authService already initialized.
+AuthService authService = mobileCore.getInstance(AuthService.class);
+UserPrincipal currentUser = authService.currentUser();
+
+authService.logout(currentUser);
+----
```

Comment:
@aidenkeating I think 1 thing we're missing in the guide is an example mobile-services.json file, or something that explains how that file relates to the auth sdk.
It may not always be the case that the user of the auth sdk is using the mobile cli to generate that file, so showing an example file might be good.